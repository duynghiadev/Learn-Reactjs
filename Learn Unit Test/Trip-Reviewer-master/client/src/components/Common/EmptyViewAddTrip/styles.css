.empty-view-container {
  padding: 25px 0 50px 0;
  text-align: center;
}
.empty-view-container .message {
  margin-bottom: 75px;
  font-size: 24px;
}
.empty-view-container .button {
  display: inline-block;
  cursor: pointer;
  font-size: 20px;
  padding: 10px 20px;
  background: #02bcd4;
  color: #fff;
  font-weight: 300;
  border-radius: 3px;
  border: none;
}
.empty-view-container .button:hover {
  background: #009eb3;
  transform: scale(1.05);
}
.empty-view-container .button:active {
  background: #0495a8;
  transform: scale(1.01);
}

@media (max-width: 400px) {
  .empty-view-container .message {
    font-size: 18px;
  }
}
