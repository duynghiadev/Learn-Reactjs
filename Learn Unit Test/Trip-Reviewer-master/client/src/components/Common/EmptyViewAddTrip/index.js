import React from 'react';
import PropTypes from 'prop-types';

import { ADD_TRIP } from '../../../constants/routes';

import './styles.css';

const EmptyViewAddTrip = ({ message, buttonText }) => (
  <div className="empty-view-container">
    <p className="message">{message}</p>
    <a className="button" href={ADD_TRIP}>
      {buttonText}
    </a>
  </div>
);

EmptyViewAddTrip.propTypes = {
  message: PropTypes.string,
  buttonText: PropTypes.string,
};

EmptyViewAddTrip.defaultProps = {
  message: 'There is no any story yet.',
  buttonText: 'Add a story',
};

export default EmptyViewAddTrip;
