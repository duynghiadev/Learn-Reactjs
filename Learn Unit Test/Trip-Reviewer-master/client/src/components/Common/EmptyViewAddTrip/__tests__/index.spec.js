import React from 'react';
import { create } from 'react-test-renderer';

import EmptyViewAddTrip from '..';

const mockComponent = props => <EmptyViewAddTrip {...props} />;

describe('<EmptyViewAddTrip />', () => {
  it('should render component with default props', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with custom props', () => {
    const props = {
      message: 'message',
      buttonText: 'buttonText',
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
