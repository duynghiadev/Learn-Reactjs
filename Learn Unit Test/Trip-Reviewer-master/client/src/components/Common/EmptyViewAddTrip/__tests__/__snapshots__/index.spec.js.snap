// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyViewAddTrip /> should render component with custom props 1`] = `
<div
  className="empty-view-container"
>
  <p
    className="message"
  >
    message
  </p>
  <a
    className="button"
    href="/trip/add"
  >
    buttonText
  </a>
</div>
`;

exports[`<EmptyViewAddTrip /> should render component with default props 1`] = `
<div
  className="empty-view-container"
>
  <p
    className="message"
  >
    There is no any story yet.
  </p>
  <a
    className="button"
    href="/trip/add"
  >
    Add a story
  </a>
</div>
`;
